src/CodeStyle/Core/Analyzers/SyntaxTreeExtensions.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Shared.Extensions
+{
+    internal static class SyntaxTreeExtensions
+    {
+        public static bool OverlapsHiddenPosition(this SyntaxTree tree, TextSpan span, CancellationToken cancellationToken)
+        {
+            if (tree == null)
+            {
+                return false;
+            }
+
+            var text = tree.GetText(cancellationToken);
+
+            return text.OverlapsHiddenPosition(span, (position, cancellationToken2) =>
+                {
+                    // implements the ASP.Net IsHidden rule
+                    var lineVisibility = tree.GetLineVisibility(position, cancellationToken2);
+                    return lineVisibility == LineVisibility.Hidden || lineVisibility == LineVisibility.BeforeFirstLineDirective;
+                },
+                cancellationToken);
+        }
```

Comment:
seemes identical to workspace version.  Not shareable?

Reply:
I intended to share this, and think I split the original file already. Will check next time I'm at the computer.